### PR TITLE
item-matrix: external relationships were not included when no relation specified

### PR DESCRIPTION
**Pull request by SteinHeselmans**
When no relations were specified in an item-matrix (=wildcard all
available relationships), the external relationships were not included.

### CODE REVIEW

**Comment by Letme** [comment]
Can we avoid codeclimate fixes or we triage them?

File: mlx/traceability.py [hunk 1]
```diff
@@ -400,18 +404,17 @@ def process_item_nodes(app, doctree, fromdocname):
...
-                for relationship in node['type']:
+                for relationship in relationships:
                     if REGEXP_EXTERNAL_RELATIONSHIP.search(relationship):
                         for target_id in source_item.iter_targets(relationship):
                             right += make_external_item_ref(app, target_id, relationship)
                 for target_id in all_item_ids:
                     target_item = env.traceability_collection.get_item(target_id)
                     # placeholders don't end up in any item-matrix (less duplicate warnings for missing items)
-                    if target_item.is_placeholder() is True:
+                    if not target_item or target_item.is_placeholder() is True:
```
Comment (Letme) [inline mlx/traceability.py]: Why is this is True needed here? Function returns True or something else (not false?)

Comment (SteinHeselmans) [reply mlx/traceability.py]: explicit true is indeed not needed, the function returns only False/True. Changed in fcd098d

**Comment by SteinHeselmans** [comment]
the codeclimate stats are very bad, so i prefer to tackle them in a future pull request. If after that the level is better, we can start maintaining this decent level.

**Pull request merged**